### PR TITLE
[4.1] Misc improvements for Task\TaskOption

### PR DESCRIPTION
**Pull request by ditsuke**
### Summary of Changes
- Add `@property-reads` docs for better IDE/static analysis support.
- Refactor `TaskOption::type` to `TaskOption::id`

### Testing Instructions
- Create a new Scheduled Task with a short interval-based schedule (1 minute).
- Let it run organically at planned time (should be a minute after created).
- Run it manually from the button in the Scheduled Tasks manager.

### Actual result BEFORE applying this Pull Request
- Everything works as expected; but:  

- TaskOption::type doesn't make sense.
- Poor IDE support for $taskOption->{property} since they're accessed only through TaskOption::__get().

### Expected result AFTER applying this Pull Request
- Everything works as expected; but:  

- TaskOption::id makes more sense.
- IDE doesn't complain about valid $taskOption->{property{ reads.

### Documentation Changes Required
\-

### CODE REVIEW

File: administrator/components/com_scheduler/src/Task/TaskOption.php [hunk 1]
```diff
@@ -20,6 +20,11 @@
  * argument pairs as argument. Internally, the TaskOption object generates the routine title and description from the
  * language constant prefix.
  *
+ * @property-read  string $desc             The routine description.
+ * @property-read  string $id               The routine ID.
+ * @property-read  string $langConstPrefix  The routine's language constant prefix.
+ * @property-read  string $title            The routine title.
+ *
  * @since  4.1.0
  */
```
Comment (PhilETaylor) [inline administrator/components/com_scheduler/src/Task/TaskOption.php]: property-read should come after since, and a blank line.

**Comment by ditsuke** [comment]
Tests, please!

**Comment by Quy** [comment]
Create a new task.

> Cannot use object of type Joomla\CMS\Object\CMSObject as array 

> JROOT\administrator\components\com_scheduler\src\Traits\TaskPluginTrait.php:214

**Comment by ditsuke** [comment]
> Create a new task.
> 
> > Cannot use object of type Joomla\CMS\Object\CMSObject as array
> 
> > JROOT\administrator\components\com_scheduler\src\Traits\TaskPluginTrait.php:214

@Quy should be resolved with 05256744d1ced2a622e79b62745b7f7d70ba156c. Requesting a retest.

**Comment by Quy** [comment]
Manually running the task.

**Comment by ditsuke** [comment]
> Manually running the task.
> 
> 

@Quy I'm unable to reproduce. Can you please share the configuration of the failing task?

Edit: I succeeding in reproducing.

**Comment by ditsuke** [comment]
@Quy apologies, my efforts to test were being hampered by an outdated JS build and changes from #36518! We should be good to go now with 56d5b7b 😅

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 56d5b7b50c5e888fc6e79fc3b4326a4686634e55<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36516">issues.joomla.org/tracker/joomla-cms/36516</a>.</sub>

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on 56d5b7b50c5e888fc6e79fc3b4326a4686634e55<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36516">issues.joomla.org/tracker/joomla-cms/36516</a>.</sub>

**Comment by jwaisner** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36516">issues.joomla.org/tracker/joomla-cms/36516</a>.</sub>

**Comment by bembelimen** [comment]
I guess this will break all 3rd party extension, using ```->type``` already, so you need to add the fallback into the magic get method (if "type" use "id") and deprecate it the same moment to make it work.

**Comment by ditsuke** [comment]
> I guess this will break all 3rd party extension, using ```->type``` already, so you need to add the fallback into the magic get method (if "type" use "id") and deprecate it the same moment to make it work.

@bembelimen added back with deprecated status.  Please review.

**Comment by richard67** [comment]
@bembelimen Does it need 2 tests again? Or is it ok to keep RTC and you review?

**Comment by bembelimen** [comment]
We should use the log class instead throwing a PHP depreciation:

https://github.com/joomla/joomla-cms/blob/4.1-dev/libraries/src/Application/CMSApplication.php#L418

Then review is enough

**Comment by richard67** [comment]
> We should use the log class instead throwing a PHP depreciation:
> 
> https://github.com/joomla/joomla-cms/blob/4.1-dev/libraries/src/Application/CMSApplication.php#L418
> 
> Then review is enough

@ditsuke Could you implement that suggestion? Thanks in advance.

**Comment by ditsuke** [comment]
@richard67 thanks for tagging. Implemented with ef13678.

**Comment by bembelimen** [comment]
Thx

**Comment by ditsuke** [comment]
thanks

**Pull request merged**